onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -1256,26 +1220,30 @@ common::Status TensorrtExecutionProvider::Compile(const std::vector<FusedNodeAnd
           return ORT_MAKE_STATUS(ONNXRUNTIME, EP_FAIL,
                                  "TensorRT EP could not build engine for fused node: " + fused_node.Name());
         }
-
-        if (engine_cache_enable_)
-          update_engine_cache = true;
-
-        // Build context
-        if (context_memory_sharing_enable_) {
-          size_t mem_size = trt_engine->getDeviceMemorySize();
```

Comment:
does this accidentally revert the context sharing changes too?

Reply:
I manually added back context sharing changes.